src/globus_sdk/paging/marker.py

```diff
@@ -34,5 +39,18 @@ def pages(self) -> Iterator[PageT]:
                 self.client_kwargs["marker"] = self.marker
             current_page = self.method(*self.client_args, **self.client_kwargs)
             yield current_page
-            self.marker = current_page.get("marker")
-            has_next_page = current_page.get("has_next_page", False)
+            self.marker = current_page.get(self.marker_key)
+            has_next_page = self._check_has_next_page(current_page)
+
+
+class NullableMarkerPaginator(MarkerPaginator[PageT]):
+    """
+    A paginator which uses `has_next_page` and `marker` from payloads, sets the `marker`
```

Comment:
Help text needs a fix here since this class explicitly does not use a `has_next_page`

Reply:
Oof! Good catch. It's a clear case of copy-paste-replace gone wrong. I'll edit this to cleanup and then we should be good to go.